### PR TITLE
Use ASP.NET Core HTTPS Development Certificate

### PR DESCRIPTION
**Pull request by mark-monteiro**
The .NET Core SDK includes an HTTPS development certificate that is is installed as part of the first-run experience ([docs](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-3.1&tabs=visual-studio#trust-the-aspnet-core-https-development-certificate-on-windows-and-macos)). When running in development mode we can serve over HTTPS using this certificate.

**Changes**
* Set `ASPNETCORE_ENVIRONMENT=Development` environment variable when running from Visual Studio, VS Code, or `dotnet run` CLI ([docs](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/environments?view=aspnetcore-3.1#development))
* At application startup, if we are in development mode, bind HTTPS ports using the SDK development certificate. If a custom certificate is configured, that is still given priority over this built-in one
* Log the values for all `DOTNET_` and `ASPNETCORE_` prefixed environment variables at server start (this includes `ASPNETCORE_ENVIRONMENT`) ([docs](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-3.1#host-configuration))

I've also added documentation about this to the project README in #2620

### CODE REVIEW

File: Jellyfin.Server/Program.cs [hunk 1]
```diff
@@ -258,6 +260,14 @@ private static IWebHostBuilder CreateWebHostBuilder(ApplicationHost appHost, ISe
                                     appHost.HttpsPort,
                                     listenOptions => listenOptions.UseHttps(appHost.Certificate));
                             }
+                            else if (builderContext.HostingEnvironment.IsDevelopment())
+                            {
+                                options.Listen(address, appHost.HttpsPort, listenOptions =>
+                                {
+                                    listenOptions.UseHttps();
+                                    listenOptions.Protocols = HttpProtocols.Http1AndHttp2;
```
Comment (mark-monteiro) [inline Jellyfin.Server/Program.cs]: **NOTE:** These ports are bound to enable HTTP/2 as outlined in #2634

**Comment by sparky8251** [comment]
Does this certificate change with install? We used to use a certificate that was identical for all installs and I don't want to return to that.

**Comment by mark-monteiro** [comment]
> Does this certificate change with install? We used to use a certificate that was identical for all installs and I don't want to return to that.

The certificate appears to be [generated dynamically](https://github.com/dotnet/aspnetcore/blob/v3.1.2/src/Shared/CertificateGeneration/CertificateManager.cs#L204).

Also, I'm not sure if this was clear from the PR description, but this will not be a feature accessible to any regular users, this certificate is only installed with the .NET SDK, not the runtime. This functionality is only meant for development and is hidden behind the config flag `ASPNETCORE_ENVIRONMENT=Development`.

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 2]
```diff
@@ -117,6 +117,11 @@ namespace Emby.Server.Implementations
     /// </summary>
     public abstract class ApplicationHost : IServerApplicationHost, IDisposable
     {
+        /// <summary>
+        /// The environment variable prefixes to log at server startup.
+        /// </summary>
+        private static readonly string[] RelevantEnvVarPrefixes = { "JELLYFIN_", "DOTNET_", "ASPNETCORE_" };
```
Comment (Bond-009) [inline Emby.Server.Implementations/ApplicationHost.cs:123]:
```suggestion
        private static readonly string[] _relevantEnvVarPrefixes = { "JELLYFIN_", "DOTNET_", "ASPNETCORE_" };
```

Comment (mark-monteiro) [reply Emby.Server.Implementations/ApplicationHost.cs]: Fixed

**Pull request merged**